composable-tasks/src/compose_tasks.rs

```diff
@@ -275,8 +277,22 @@ pub trait TaskCall:
         self.smart_swap_fees(&token_out.clone().unwrap_esdt())
             .update(|total_fees| *total_fees += &fee_taken);
 
-        let remaining_amount_after_fee = amount_out - fee_taken;
-        EgldOrEsdtTokenPayment::new(token_out, 0, remaining_amount_after_fee)
+        let remaining_amount_after_fee = amount_out - fee_taken.clone();
+
+        let payment_out = EgldOrEsdtTokenPayment::new(token_out, 0, remaining_amount_after_fee);
+
+        self.emit_smart_swap_event(
```

Comment:
Just to keep it more clear, create the variable before you call the event. That way you keep the same approach as with the payment_out variable.